manticore/platforms/linux.py

```diff
@@ -1225,15 +1247,13 @@ def sys_fsync(self, fd):
 
         ret = 0
         try:
-            f = self.files[fd]
-            if isinstance(f, Socket):
-                ret = -errno.EINVAL
-            else:
-                f.sync()
+            self.files[fd].sync()
         except IndexError:
             ret = -errno.EBADF
+        except BadFd:
```

Comment:
Yep, I was confused a bit by the IndexError comment at https://github.com/trailofbits/manticore/pull/301/files/89afa2325a62db323730cee42c1481201c55c593

Ok so we raise Badfd a and that should do it